frontend/packages/vsphere-plugin/src/components/persist.ts

```diff
@@ -183,43 +197,49 @@ const updateIniFormat = (
     initValues.defaultDatastore || '/datacenterplaceholder/datastore/defaultdatastoreplaceholder';
   const initFolder = initValues.folder || '/datacenterplaceholder/vm/folderplaceholder';
 
-  cfg = findAndReplace(
-    cfg,
+  let result: UpdateConfigMapResult = { config: cfg, expectedValues: [] };
+
+  result = getUpdatedConfig(
+    result,
     `[VirtualCenter "${initVCenter}"]`,
     `[VirtualCenter "${values.vcenter}"]`,
   );
 
-  cfg = findAndReplace(cfg, `server = "${initVCenter}"`, `server = "${values.vcenter}"`);
-  cfg = findAndReplace(
-    cfg,
+  result = getUpdatedConfig(result, `server = "${initVCenter}"`, `server = "${values.vcenter}"`);
+  result = getUpdatedConfig(
+    result,
     `datacenters = "${initDatacenter}"`,
     `datacenters = "${values.datacenter}"`,
   );
-  cfg = findAndReplace(
-    cfg,
+  result = getUpdatedConfig(
+    result,
     `datacenter = "${initDatacenter}"`,
     `datacenter = "${values.datacenter}"`,
   );
-  cfg = findAndReplace(
-    cfg,
+  result = getUpdatedConfig(
+    result,
     `default-datastore = "${initDatastore}"`,
     `default-datastore = "${values.defaultDatastore}"`,
   );
-  cfg = findAndReplace(cfg, `folder = "${initFolder}"`, `folder = "${values.folder}"`);
-  cfg = findAndReplace(
-    cfg,
+  result = getUpdatedConfig(result, `folder = "${initFolder}"`, `folder = "${values.folder}"`);
+  result = getUpdatedConfig(
+    result,
     `resourcepool-path = "/${initDatacenter}/host/${initVCenterCluster}/Resources"`,
     `resourcepool-path = "/${values.datacenter}/host/${values.vCenterCluster}/Resources"`,
   );
 
-  if (!cfg) {
+  if (result.expectedValues.length > 0) {
     throw new PersistError(
-      t('Failed to parse cloud provider config {{cm}}', { cm: cloudProviderConfig.metadata.name }),
-      t('Unknown format'),
+      t('Failed to parse cloud provider config {{cm}}. ', {
```

Comment:
the extra space and punctuation in this string is causing duplicate strings to appear in the locales json file

```suggestion
      t('Failed to parse cloud provider config {{cm}}', {
```